addons/web_editor/controllers/main.py

```diff
@@ -545,3 +547,43 @@ def subber(match):
             ('Content-type', 'image/svg+xml'),
             ('Cache-control', 'max-age=%s' % http.STATIC_CACHE_LONG),
         ])
+
+    @http.route(['/web_editor/media_library_url'], type='json', auth="user", website=True)
+    def media_library_url(self):
+        dbuuid = request.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        endpoint = request.env['ir.config_parameter'].sudo().get_param('web_editor.media_library_endpoint', DEFAULT_LIBRARY_ENDPOINT)
+        return '%s?dbuuid=%s' % (endpoint, dbuuid)
+
+    @http.route('/web_editor/save_media', type='json', auth='user', methods=['POST'])
+    def save_media(self, media=[]):
+        """
+        Saves images from the media library as new attachments, making them
```

Comment:
@mart-e 
What's the last security rule for SVG ?

Reply:
> Olivier (odo): - si c'est stocké dans un ir.attachment, lors du create() on verifie que user._is_system() et on force le mimetype en text/plain (ce qui en theorie casserait les SVG)
> 
> Olivier (odo): - si c'est stocké dans un fields qui n'est pas en attachment, alors on check record.env.is_system() et si c'est pas le cas on raise une error
> 

=> donc ca ne doit pas marché quand t'es pas admin dans l'etat.

Reply:
Indeed, just tried to edit as a designer and it failed, attachment did come back as `text/plain`. Should I check the urls against a white-list of trusted origins and create the attachments with sudo?

Reply:
we could imagine to bypass it for website_designer but quid in all others cases where end user will try...

Probably that just whitelist our media library and make sudo is a possibility
+ request of odony is to avoid call from browser to IAP in this case, but make it s2s directly and avoid to leak the dbuuid.